src/utils.js

```diff
@@ -329,22 +336,16 @@ export const changePin = async (wallet, oldPin, newPin) => {
 };
 
 /**
- * Map history element to expected TxHistory model object
+ * Curry function that maps a raw history element to an expected TxHistory model object.
  *
- * element {Object} Tx history element with {txId, timestamp, balance, voided?}
- * token {string} Token uid
+ * @param {string} tokenUid - Token uid
+ *
+ * @returns {(rawTxHistory: Object) => TxHistory} A function that maps a raw
+ * transaction history element to a TxHistory object
  */
-export const mapTokenHistory = (element, token) => {
-  const data = {
-    txId: element.txId,
-    timestamp: element.timestamp,
-    balance: element.balance,
-    // in wallet service this comes as 0/1 and in the full node comes with true/false
-    isVoided: Boolean(element.voided),
-    tokenUid: token
-  };
-  return new TxHistory(data);
-};
+export const mapToTxHistory = (tokenUid) => (rawTxHistory) => (
```

Comment:
What's the benefit of having this instead of `export const mapToTxHistory = (tokenUid, rawTxHistory)`?

Reply:
Conciseness. I think this is simple to understand and improve readability.